packages/api/src/platforms/vtex/resolvers/query.ts

```diff
@@ -478,19 +478,31 @@ export const Query = {
         },
       }
     } catch (error) {
-      const result = JSON.parse((error as Error).message).error as {
-        code: string
-        message: string
-        exception: any
+      const errorMessage = (error as Error).message
+
+      let result: {
+        code?: string
+        message?: string
+        exception?: any
+      } = {}
+
+      try {
+        const parsed = JSON.parse(errorMessage)
+        result = parsed.error || parsed
+      } catch {
+        result = { message: errorMessage }
       }
 
-      if (result?.message?.toLowerCase()?.includes('order not found')) {
-        throw new NotFoundError(`No order found for id ${orderId}`)
+      const message =
+        result?.message?.toLowerCase() || errorMessage.toLowerCase()
+
+      if (message.includes('order not found')) {
+        throw new NotFoundError(`No order found for id ${orderId}. ${message}`)
       }
 
-      if (result?.message?.toLowerCase()?.includes('acesso negado')) {
+      if (message.includes('acesso negado')) {
```

Comment:
question: could we check the code instead of the message? I believe the message is easier to change over time.
Or at least try the code first and if it don't match, check the message